src/spu/src/services/public/stream_fetch.rs

```diff
@@ -169,96 +169,86 @@ impl StreamFetchHandler {
         let mut leader_offset_receiver = self.leader_state.offset_listener(&self.isolation);
 
         let mut counter: i32 = 0;
-        let mut consumer_offset: Option<Offset> = if !consumer_wait {
-            // since we don't need to wait for consumer, can move consumer to same offset as last read
-            Some(last_read_offset)
-        } else {
-            None // offset for consumer
-        };
+        // since we don't need to wait for consumer, can move consumer to same offset as last read
+        let mut consumer_offset: Option<Offset> = (!consumer_wait).then(|| last_read_offset);
 
         loop {
             counter += 1;
-
             debug!(counter, ?consumer_offset, last_read_offset, "waiting");
 
             select! {
-
                 _ = self.end_event.listen() => {
                     debug!("end event has been received, terminating");
                     break;
                 },
 
-
                 changed_consumer_offset = self.consumer_offset_listener.listen() => {
+                    if changed_consumer_offset == INIT_OFFSET {
+                        continue;
+                    }
 
+                    if changed_consumer_offset >= last_read_offset {
+                        debug!(
+                            changed_consumer_offset,
+                            last_read_offset,
+                            "consume caught up"
+                        );
+                        consumer_offset = Some(changed_consumer_offset);
+                        continue;
```

Comment:
Again reducing rightward-drift by using `continue`.

This is the `else` body from https://github.com/infinyon/fluvio/pull/1188/files#diff-5bcc5ec4673d6a5c8633195a7360c3402f873769c1faa90cf627b242179d2241L217-L222

Reply:
not if this is clearer.  I suggest leave this alone.  If you want to update this, I think need unit test to cover